connector-agent/src/data_sources/postgres.rs

```diff
@@ -1,15 +1,133 @@
-// use super::DataSource;
-// use crate::errors::Result;
-// use crate::types::TypeInfo;
-// use r2d2::PooledConnection;
-// use r2d2_postgres::{postgres::NoTls, PostgresConnectionManager};
-
-// impl DataSource for PooledConnection<PostgresConnectionManager<NoTls>> {
-//     fn run_query(&mut self, query: &str) -> Result<()> {
-//         unimplemented!()
-//     }
-
-//     fn produce<T: TypeInfo>(&mut self) -> Result<T> {
-//         unimplemented!()
-//     }
-// }
+use crate::data_order::DataOrder;
+use crate::data_sources::{DataSource, Produce, SourceBuilder};
+use crate::errors::{ConnectorAgentError, Result};
+use crate::types::DataType;
+use fehler::throw;
+use r2d2::{Pool, PooledConnection};
+use r2d2_postgres::{postgres::NoTls, PostgresConnectionManager};
+use std::io::Read;
+
+type PgManager = PostgresConnectionManager<NoTls>;
+type PgConn = PooledConnection<PgManager>;
+
+pub struct PostgresDataSourceBuilder {
+    pool: Pool<PgManager>,
+    data_order: Option<DataOrder>,
```

Comment:
Is this data_order used anywhere? If not maybe we can remove it?

Reply:
Thanks Xiaoying! Will look into this.